File: Cpp/Aksesoris.cpp
#include <bits/stdc++.h>
#include "PetShop.cpp"
using namespace std;

class Aksesoris : public PetShop {
private:
    string jenis;
    string bahan;
    string warna;

public:
    // Konstruktor Default
    Aksesoris() : PetShop() {
        this->jenis = "";
        this->bahan = "";
        this->warna = "";
    }

    // Konstruktor dengan Parameter
    Aksesoris(int id, string nama, string kategori, double harga, string jenis, string bahan, string warna) 
        : PetShop(id, nama, kategori, harga) {
        this->jenis = jenis;
        this->bahan = bahan;
        this->warna = warna;
    }

    // Getter dan Setter (dengan const pada getter)
    string get_jenis() const { return this->jenis; }
    void set_jenis(string jenis) { this->jenis = jenis; }
    string get_bahan() const { return this->bahan; }
    void set_bahan(string bahan) { this->bahan = bahan; }
    string get_warna() const { return this->warna; }
    void set_warna(string warna) { this->warna = warna; }

    // Destructor
    ~Aksesoris() {}
};

File: Java/Aksesoris.java
// Aksesoris.java
// Kelas Aksesoris merupakan turunan dari PetShop dan menambahkan atribut khusus

// Pastikan PetShop.java berada dalam paket yang sama (default package)
public class Aksesoris extends PetShop {
    // Deklarasi atribut tambahan untuk aksesoris
    protected String jenis; // Jenis aksesoris
    protected String bahan; // Bahan aksesoris
    protected String warna; // Warna aksesoris

    // Konstruktor default
    public Aksesoris() { // Constructor tanpa parameter
        super(); // Panggil konstruktor default PetShop
        this.jenis = ""; // Inisialisasi jenis dengan string kosong
        this.bahan = ""; // Inisialisasi bahan dengan string kosong
        this.warna = ""; // Inisialisasi warna dengan string kosong
    }

    // Konstruktor dengan parameter
    public Aksesoris(int id, String namaProduk, String kategori, double harga, String jenis, String bahan,
            String warna) {
        super(id, namaProduk, kategori, harga); // Panggil konstruktor PetShop dengan parameter
        this.jenis = jenis; // Set jenis aksesoris
        this.bahan = bahan; // Set bahan aksesoris
        this.warna = warna; // Set warna aksesoris
    }

    // Getter untuk jenis
    public String getJenis() {
        return this.jenis; // Mengembalikan jenis aksesoris
    }

    // Setter untuk jenis
    public void setJenis(String jenis) {
        this.jenis = jenis; // Mengatur jenis aksesoris
    }

    // Getter untuk bahan
    public String getBahan() {
        return this.bahan; // Mengembalikan bahan aksesoris
    }

    // Setter untuk bahan
    public void setBahan(String bahan) {
        this.bahan = bahan; // Mengatur bahan aksesoris
    }

    // Getter untuk warna
    public String getWarna() {
        return this.warna; // Mengembalikan warna aksesoris
    }

    // Setter untuk warna
    public void setWarna(String warna) {
        this.warna = warna; // Mengatur warna aksesoris
    }
}
